pkg/build/builder/common.go

```diff
@@ -46,8 +46,10 @@ func buildInfo(build *api.Build) []KeyValue {
 	}
 	if build.Spec.Source.Git != nil {
 		sourceURL := build.Spec.Source.Git.URI
-		if originalURL, ok := build.Annotations[OriginalSourceURLAnnotationKey]; ok {
-			sourceURL = originalURL
+		if build.Annotations != nil {
```

Comment:
A nil check is not needed when reading from a map. The second return value will be false if the map is nil or if the map does not contain the given key